### PR TITLE
feat(core): add Atlassian Statuspage check adapter

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

This adds an Atlassian Statuspage check adapter to the status-please monorepo (a Bun + TypeScript upptime successor). It lets a configured `site` mirror status directly from any Atlassian Statuspage (status.claude.com, *.statuspage.io, vercel-status.com, etc.) via the standard `/api/v2/summary.json` JSON API, either for the whole page's overall indicator or for a single named/id'd component. `checkSite` was refactored into a dispatcher on `site.check`; the new adapter maps the page's overall indicator — or a single configured `component` (by name or id) — to the up/degraded/down model. Only the base page URL is needed; the summary.json path is derived automatically.

Changed files:
- `packages/core/src/config.ts` — `'statuspage'` kind added to `checkKindSchema`, optional `component` field
- `packages/core/src/check.ts` — `checkSite` refactored into a dispatcher; added `checkStatuspage`, `deriveStatuspageStatus`, `statuspageSummaryUrl`
- `packages/core/src/check.test.ts` — 13 new tests
- `status.config.example.yml` — statuspage usage examples added
- `README.md` — statuspage config docs added

## Related issue

N/A — no related issue for this PR.

## Checklist

- [x] PR title follows Conventional Commits
- [x] Tests added or updated, and the suite passes (`bun run test`)
- [x] Lint/format pass (`bun run lint`)
- [x] Documentation updated if behavior changed
- [x] No breaking change, or a `BREAKING CHANGE:` note is included

All 98 tests pass (`bun run test`), typecheck and lint are clean, and the adapter was verified end-to-end against live status.claude.com and vercel-status.com endpoints.

---
## Summary by cubic
Adds a `statuspage` check that mirrors health from Atlassian Statuspage `/api/v2/summary.json` for a whole page or a single component. Refactors `checkSite` in `packages/core` to dispatch by `site.check`, validates payloads with a `zod` schema, preserves real HTTP codes, and rejects `component` on non-`statuspage` checks at config parse.

- **New Features**
  - Support `check: statuspage` in `packages/core` with optional `component` (name or id); API URL auto-derived via `statuspageSummaryUrl`.
  - Docs: README “Check types” table and full adapter guide at `docs/adapters/statuspage.md`. Examples and tests cover URL derivation, component matching, and schema/edge cases.

- **Bug Fixes**
  - Preserve real HTTP status for grading failures; differentiate network errors (`code: 0`) from payload/config issues.
  - Validation and edges: `zod`-validate `summary.json` (including non-object/invalid shapes); strip trailing slashes before detecting `/api/v2/*.json`; handle explicit `/api/v2/*.json` with trailing slashes; avoid ReDoS in trailing-slash stripping; trim `component` before id/name match; reject `component` on non-`statuspage` checks with a clear parse error.

<sup>Written for commit 10cd0d2ecda896bfe1a4dd284772c4e0f4166dd3. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/core/src/check.ts [hunk 1]
```diff
@@ -63,3 +72,104 @@ export async function checkSite(
...
+  major: 'down',
+  critical: 'down',
+  maintenance: 'degraded',
+}
+
+/** The slice of an Atlassian Statuspage `summary.json` payload we rely on. */
+export interface StatuspageSummary {
+  status?: { indicator?: string, description?: string }
+  components?: Array<{ id?: string, name?: string, status?: string }>
+}
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts]: @gemini-code-assist 좋은 지적 감사합니다. `statuspageSummarySchema`(zod)를 정의하고 `StatuspageSummary`를 `z.infer`로 도출하도록 적용했습니다. `import { z }`도 파일 상단으로 이동했습니다. (커밋 bdc4995)

File: packages/core/src/check.ts [hunk 2]
```diff
@@ -63,3 +72,104 @@ export async function checkSite(
...
+      return {
+        slug: site.slug,
+        status: 'down',
+        code: res.status,
+        responseTime,
+        checkedAt: new Date(start).toISOString(),
+        error: `Statuspage API returned ${res.status}`,
+      }
+    }
+    const summary = (await res.json()) as StatuspageSummary
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts]: @gemini-code-assist 반영했습니다. `res.json()` 결과를 `statuspageSummarySchema.safeParse(...)`로 경계에서 검증하고, 실패 시 실제 HTTP status를 보존한 채 `down`으로 기록합니다. safeParse라 null·배열 등 잘못된 형태도 함께 걸러집니다. (커밋 bdc4995)

File: packages/core/src/check.ts [hunk 3]
```diff
@@ -63,3 +72,104 @@ export async function checkSite(
...
+ * Resolve the `summary.json` API URL for a configured Statuspage `url`. A bare
+ * page URL (`https://status.claude.com`) gets `/api/v2/summary.json` appended;
+ * a URL already pointing at an `/api/v2/*.json` endpoint is used verbatim.
+ */
+export function statuspageSummaryUrl(url: string): string {
+  if (/\/api\/v2\/[^/]+\.json$/.test(url)) {
+    return url
+  }
+  return `${url.replace(/\/+$/, '')}/api/v2/summary.json`
+}
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts:142]: @gemini-code-assist 제안 감사합니다. 다만 이 변경은 적용하지 않았습니다. 제안 스니펫의 `parsed.pathname.replace(/\/+$/, '')`가 직전 커밋(0f5df4e)에서 CodeQL `js/polynomial-redos`(high) 때문에 제거한 ReDoS 정규식을 그대로 재도입합니다. 또한 쿼리스트링 케이스도 실제로는 해결되지 않습니다 — 원인은 얼리리턴 가드가 `\.json$`로 앵커되어 `?query`가 붙으면 매칭 실패하는 것이라 `new URL`을 써도 pathname에 경로가 이중으로 붙습니다. `url`은 커밋된 `status.config.yml`에서 오는 신뢰 입력이고 `z.string().url()`로 검증되며, 현재 trailing-slash 제거는 백트래킹 없는 선형 스캔이라 ReDoS-safe합니다. 이런 이유로 현행 구현을 유지합니다.

File: packages/core/src/check.ts [hunk 4]
```diff
@@ -63,3 +73,153 @@ export async function checkSite(
...
+  }
+  // Strip trailing slashes with a linear character scan rather than a
+  // backtracking regex (`/\/+$/` is a polynomial-ReDoS pattern on inputs with
+  // many trailing slashes — flagged by CodeQL js/polynomial-redos).
+  let end = url.length
+  while (end > 0 && url.charCodeAt(end - 1) === 47 /* '/' */) {
+    end--
+  }
+  return `${url.slice(0, end)}/api/v2/summary.json`
+}
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts:142]: @gemini-code-assist 정확한 버그입니다. trailing slash를 먼저 제거한 뒤 엔드포인트 패턴을 검사하도록 순서를 변경해 explicit `/api/v2/*.json` + trailing slash 케이스의 경로 중복을 해결했습니다. ReDoS-safe한 선형 스캔은 유지했습니다. (커밋 c824ea1)

File: packages/core/src/check.ts [hunk 5]
```diff
@@ -63,3 +73,153 @@ export async function checkSite(
...
+ * page's overall `indicator` is used. Unknown status strings map to `degraded`
+ * (something is off, but not clearly an outage). Throws when a named component
+ * isn't present so the caller records it as `down` with a clear error.
+ */
+export function deriveStatuspageStatus(summary: StatuspageSummary, component?: string): CheckStatus {
+  if (component !== undefined) {
+    const target = component.trim().toLowerCase()
+    const match = summary.components?.find(
+      c => c.id === component || c.name?.trim().toLowerCase() === target,
+    )
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts:157]: @gemini-code-assist 반영했습니다. id 비교에도 `component.trim()`을 적용해 name 매칭과 공백 처리를 일관되게 맞췄습니다. 회귀 테스트도 추가했습니다. (커밋 c824ea1)

File: packages/core/src/check.test.ts [hunk 6]
```diff
@@ -40,3 +56,168 @@ describe('checkSite', () => {
...
+
+  it('strips multiple trailing slashes', () => {
+    expect(statuspageSummaryUrl('https://status.claude.com///'))
+      .toBe('https://status.claude.com/api/v2/summary.json')
+  })
+
+  it('leaves an explicit api/v2 endpoint untouched', () => {
+    const url = 'https://status.claude.com/api/v2/status.json'
+    expect(statuspageSummaryUrl(url)).toBe(url)
+  })
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.test.ts:79]: @gemini-code-assist 추가했습니다. trailing slash가 포함된 explicit API URL 케이스에 대한 회귀 테스트를 넣었습니다. (커밋 c824ea1)

File: packages/core/src/config.ts [hunk 7]
```diff
@@ -24,6 +24,12 @@ export const siteSchema = z
     expectedStatusCodes: z.array(z.number().int()).default([200]),
     /** Response time above this (ms) marks the site `degraded` rather than `up`. */
     maxResponseTime: z.number().int().positive().default(5000),
+    /**
+     * For `check: statuspage` only. Reads one Atlassian Statuspage component by
+     * name (case-insensitive) or id; when omitted, the page's overall indicator
+     * is used. Ignored by other check kinds.
+     */
+    component: z.string().min(1).optional(),
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/config.ts:33]: @greptile-apps 좋은 지적입니다. `component`가 non-statuspage check kind에 붙으면 parse 단계에서 에러를 내도록 zod `superRefine` 교차검증을 추가했습니다. 오타(예: `check: http` + `component:`)가 런타임에 조용히 무시되지 않고 즉시 드러납니다. 문서와 테스트도 갱신했습니다. (커밋 3b34aaf)

File: packages/core/src/check.ts [hunk 8]
```diff
@@ -63,3 +73,158 @@ export async function checkSite(
...
+  // many trailing slashes — flagged by CodeQL js/polynomial-redos). Testing the
+  // cleaned URL against the endpoint pattern also handles an explicit
+  // `/api/v2/*.json` URL that carries a trailing slash (otherwise the guard
+  // would miss it and produce a doubled `.../summary.json/api/v2/summary.json`).
+  let end = url.length
+  while (end > 0 && url.charCodeAt(end - 1) === 47 /* '/' */) {
+    end--
+  }
+  const cleaned = url.slice(0, end)
+  if (/\/api\/v2\/[^/]+\.json$/.test(cleaned)) {
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/core/src/check.ts:138]: @gemini-code-assist 검토했지만 이 건은 적용하지 않겠습니다. Atlassian Statuspage의 API 경로는 항상 소문자(`/api/v2/summary.json`)로 제공되어 `.JSON` 같은 대문자 입력은 실제로 발생하지 않습니다. 또한 `i` 플래그로 대문자 경로를 '이미 API 엔드포인트'로 인정하더라도, 그 URL을 그대로 fetch하면 소문자만 서빙하는 Statuspage에서 404가 나므로 문제(더블 append)가 404로 바뀔 뿐 실제로 해결되지는 않습니다. `url`은 커밋된 `status.config.yml`에서 오는 신뢰 입력이고 `z.string().url()`로 검증됩니다. 이런 이유로 현행 소문자 매칭을 유지합니다.
